File: ROT47.cpp
#include <iostream>

using namespace std;

void cipherEncryption(){
    string message;
    cout << "Enter message: ";
    getline(cin, message);
    cin.ignore();

    int key = 47;
    string encrypText = "";

    for (int i = 0; i < message.length(); i++){
        int temp = message[i] + key;
        if(message[i] == 32){
            encrypText += " ";
        } else if (temp > 126){
            temp -= 94;
            encrypText += temp;
        } else {
            encrypText += temp;
        } // if-else
    } // for

    cout << "Encrypted Text: " << encrypText;
}

void cipherDecryption(){
    string message;
    cout << "Enter message: ";
    getline(cin, message);
    cin.ignore();

    int key = 47;
    string decrypText = "";

    for (int i = 0; i < message.length(); i++){
        int temp = message[i] - key;
        if(message[i] == 32){
            decrypText += " ";
        } else if (temp < 32){
            temp += 94;
            decrypText += temp;
        } else {
            decrypText += temp;
        } // if-else
    } // for

    cout << "Decrypted Text: " << decrypText;
}

int main()
{
    cout << "1. Encryption\n2. Decryption\nChoose(1,2): ";
    int choice;
    cin >> choice;
    cin.ignore();

    if (choice == 1){
        cout << endl << "---Encryption---" << endl;
        cipherEncryption();

    } else if (choice == 2){
        cout << endl << "---Decryption---" << endl;
        cipherDecryption();

    } else {
        cout << "Invalid Choice";
    }

    return 0;
}

File: ROT47.py
def cipher_encryption():
    message = input("Enter message: ")
    key = 47
    encryp_text = ""

    for i in range(len(message)):
        temp = ord(message[i]) + key
        if ord(message[i]) == 32:
            encryp_text += " "
        elif temp > 126:
            temp -= 94
            encryp_text += chr(temp)
        else:
            encryp_text += chr(temp)
        # if-else
    # for

    print("Encrypted Text: {}".format(encryp_text))


def cipher_decryption():
    message = input("Enter message: ")
    key = 47
    decryp_text = ""

    for i in range(len(message)):
        temp = ord(message[i]) - key
        if ord(message[i]) == 32:
            decryp_text += " "
        elif temp < 32:
            temp += 94
            decryp_text += chr(temp)
        else:
            decryp_text += chr(temp)
            # if-else
    # for

    print("Decrypted Text: {}".format(decryp_text))


def main():
    choice = int(input("1. Encryption\n2. Decryption\nChoose(1,2): "))
    if choice == 1:
        print("---Encryption---")
        cipher_encryption()
    elif choice == 2:
        print("---Decryption---")
        cipher_decryption()
    else:
        print("Invalid Choice")

if __name__ == "__main__":
    main()
